File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    count_map = {}
    for key in keymap:
        for i,k in enumerate(key):
            if k in count_map:
                if i + 1 < count_map[k]:
                    count_map[k] = i + 1
            else:
                count_map[k] = i + 1
                
    for target in targets:
        available = True 
        count = 0
        for t in target:
            if t in count_map:
                count += count_map[t]
            else:
                answer.append(-1)
                available = False
                break
        if available:
            answer.append(count)
        
        
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    const key = {};
    for (let i = 0; i < keymap.length; i++) {
        const pad = keymap[i]
        for (let j = 0; j < pad.length; j++) {
            const word = pad[j]
            if (word in key) {
                key[word] = Math.min(key[word], j + 1)
            } else {
                key[word] = j + 1
            }
        }
    }
    
    for (let i = 0; i < targets.length; i++) {
        let count = 0;
        for (let j = 0; j < targets[i].length; j++) {
            const word = targets[i][j]
            if (word in key) count += key[word]
            else {
                count = -1
                break
            }
        }
        answer.push(count)
    }
    
    return answer;
}